pyiceberg/table/snapshots.py

```diff
@@ -247,9 +248,12 @@ def __str__(self) -> str:
         result_str = f"{operation}id={self.snapshot_id}{parent_id}{schema_id}"
         return result_str
 
-    def manifests(self, io: FileIO) -> List[ManifestFile]:
-        if self.manifest_list is not None:
-            file = io.new_input(self.manifest_list)
+    @staticmethod
+    @lru_cache
+    def manifests(io: FileIO, manifest_list: str) -> List[ManifestFile]:
+        """Return the manifests for the given snapshot."""
+        if manifest_list not in (None, ""):
+            file = io.new_input(manifest_list)
```

Comment:
How about keeping the original interface while adding a helper that using lru_cache?
```python
@staticmethod
    @lru_cache
    def _manifests(io: FileIO, manifest_list: str) -> List[ManifestFile]:
        if manifest_list not in (None, ""):
            file = io.new_input(manifest_list)
            return list(read_manifest_list(file))
        return []

    def manifests(self, io: FileIO) -> List[ManifestFile]:
        return self._manifests(io, self.manifest_list)
```
`manifests` is a public API so we'd better not to change it unless necessary

Reply:
This is indeed a better solution, I'll update the PR